zerver/webhooks/facebook/view.py

```diff
@@ -0,0 +1,123 @@
+            )
+    return message
+
+def page_object(payload: Dict[Text, Any]) -> Text:
+    field = payload['entry']['changes']['field']
+
+    message = ''
+    if field == 'conversations':
+        message = message + 'Updated **conversations**'
+        message = message + '\n[Open conversations...](https://www.facebook.com/'\
+                            '{page_id}/{thread_id})'.format(
+                                page_id=payload['entry']['changes']['value']['page_id'],
+                                thread_id=payload['entry']['changes']['value']['thread_id']
+                            )
+    elif field == 'website':
+        message = message + 'Changed **website**'
+    return message
+
+def permissions_object(payload: Dict[Text, Any]) -> Text:
+    field = payload['entry']['changes']['field']
+    message = '**{field} permission** changed'.format(field=field)
+    if field == 'ads_management':
+        message = message + '\n* {verb}'.format(
+            verb=payload['entry']['changes']['value']['verb']
+        )
+        for id in payload['entry']['changes']['value']['target_ids']:
+            message = message + '\n  * {id}'.format(id=id)
+    elif field == 'manage_pages':
+        message = message + '\n* {verb}'.format(
+            verb=payload['entry']['changes']['value']['verb']
+        )
+        for id in payload['entry']['changes']['value']['target_ids']:
+            message = message + '\n  * {id}'.format(id=id)
+    return message
+
+def application_object(payload: Dict[Text, Any]) -> Text:
+    field = payload['entry']['changes']['field']
+    message = '**{field}** received'.format(field=field)
+    if field == 'plugin_comment':
+        message = message + '\n**{msg_user}:**\n```quote\n{message}\n```'.format(
+            msg_user=payload['entry']['changes']['value']['from']['name'],
+            message=payload['entry']['changes']['value']['message']
+        )
+    if field == 'plugin_comment_reply':
+        message = message + '\n**{prt_msg_user}:** (Parent)\n'\
+            '```quote\n{prt_message}\n```'.format(
+                prt_msg_user=payload['entry']['changes']['value']['parent']['from']['name'],
+                prt_message=payload['entry']['changes']['value']['parent']['message']
+            )
+        message = message + '\n**{cld_msg_user}:**\n```quote\n'\
+            '```quote\n{cld_message}\n```\n```'.format(
+                cld_msg_user=payload['entry']['changes']['value']['from']['name'],
+                cld_message=payload['entry']['changes']['value']['message']
+            )
+    return message
+
+@api_key_only_webhook_view("Facebook")
+@has_request_variables
+def api_facebook_webhook(request: HttpRequest, user_profile: UserProfile,
+                         stream: Text=REQ(default='Facebook'), token: Text=REQ(),
+                         payload: Dict[Text, Any]=REQ(argument_type='body')) -> HttpResponse:
+    tmp_body = str(payload)
```

Comment:
Okay, first of all, you don't need to convert `payload` to a `str`. `payload` is a dictionary and I would recommend leaving it as it is and dealing with it directly.

Reply:
Also, if you don't convert it to a `str`, you won't have to call `_payload = json.loads(tmp_body)` either. So remove that too.